openslides/core/static/js/core/base.js

```diff
@@ -701,24 +702,26 @@ angular.module('OpenSlidesApp.core', [
                             {"action": action, "direction": direction}
                     );
                 },
-                getStateForCurrentSlide: function () {
+                getFormForCurrentSlide: function () {
                     var return_dict;
                     angular.forEach(this.elements, function(value, key) {
                         if (value.name == 'agenda/list-of-speakers') {
                             return_dict = {
-                                'state': 'agenda.item.detail',
-                                'param': {id: value.id}
+                                form: $injector.get('TopicForm'),
+                                id: value.id,
                             };
                         } else if (
                             value.name != 'agenda/item-list' &&
                             value.name != 'core/clock' &&
                             value.name != 'core/countdown' &&
                             value.name != 'core/projector-message' &&
                             value.name != 'agenda/current-list-of-speakers' ) {
-                            return_dict = {
-                                'state': value.name.replace('/', '.')+'.detail.update',
-                                'param': {id: value.id}
-                            };
+                                var formName = value.name.split('/')[1];
+                                formName = formName.charAt(0).toUpperCase() + formName.slice(1) + 'Form';
```

Comment:
Here the same. Maybe we need a factory that transforms collection names in CamelCase angular stuff.